compiler/src/iree/compiler/Codegen/SPIRV/SPIRVMapMemRefStorageClass.cpp

```diff
@@ -47,14 +47,23 @@ Optional<spirv::StorageClass> mapHALDescriptorTypeForVulkan(Attribute attr) {
 
 Optional<spirv::StorageClass> mapHALDescriptorTypeForOpenCL(Attribute attr) {
   auto dtAttr = attr.dyn_cast_or_null<IREE::HAL::DescriptorTypeAttr>();
-  if (!dtAttr) return spirv::mapMemorySpaceToOpenCLStorageClass(attr);
-  switch (dtAttr.getValue()) {
-    case IREE::HAL::DescriptorType::UniformBuffer:
-      return spirv::StorageClass::Uniform;
-    case IREE::HAL::DescriptorType::StorageBuffer:
-      return spirv::StorageClass::CrossWorkgroup;
+  if (dtAttr) {
```

Comment:
While we are at here, could you actually update both the Vulkan case and here to do `if (auto dtAttr = ...) { ... }`? Thanks!

Reply:
Sure